Controller/Adminhtml/ImageLabel/AbstractUpload.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ *
+ * @category  Smile
+ * @package   Smile\ProductLabel
+ * @author    Houda EL RHOZLANE <hoelr@smile.fr>
+ * @copyright 2019 Smile
+ */
+
+namespace Smile\ProductLabel\Controller\Adminhtml\ImageLabel;
+
+use Magento\Framework\Controller\ResultFactory;
+
+/**
+ * Class AbstractUpload
+ */
+abstract class AbstractUpload extends \Magento\Backend\App\Action
+{
+    /**
+     * Image uploader
+     *
+     * @var \Magento\Catalog\Model\ImageUploader
+     */
+    protected $imageUploader;
+
+    /**
+     * AbstractUpload constructor.
+     *
+     * @param \Magento\Backend\App\Action\Context  $context
+     * @param \Magento\Catalog\Model\ImageUploader $imageUploader
+     */
+    public function __construct(
+        \Magento\Backend\App\Action\Context $context,
+        \Magento\Catalog\Model\ImageUploader $imageUploader
+    ) {
+        parent::__construct($context);
+        $this->imageUploader = $imageUploader;
+    }
+
+    /**
+     * @return string
+     */
+    abstract protected function getFileId();
+
+    /**
+     * Upload file controller action
+     *
+     * @return \Magento\Framework\Controller\ResultInterface
+     */
+    public function execute()
+    {
+        try {
```

Comment:
Why using cookies here ? It did not work without ?

Reply:
No, it works, i removed it. 
Done.